src/routes/videoPlaybackProxy.ts

```diff
@@ -92,47 +94,88 @@ videoPlaybackProxy.get("/", async (c) => {
 
     const fetchClient = await getFetchClient(config);
 
+    let headResponse: Response | undefined;
     let location = `https://${host}/videoplayback?${queryParams.toString()}`;
 
     // https://datatracker.ietf.org/doc/html/draft-ietf-httpbis-p2-semantics-17#section-7.3
     // A maximum of 5 redirections is defined in the note of the section 7.3
     // of this RFC, that's why `i < 5`
     for (let i = 0; i < 5; i++) {
-        const googlevideoResponse: Response = await fetchClient.call(
-            undefined,
-            location,
-            {
-                method: "HEAD",
-                headers: headersToSend,
-                redirect: "manual",
-            },
-        );
+        const googlevideoResponse: Response = await fetchClient(location, {
+            method: "HEAD",
+            headers: headersToSend,
+            redirect: "manual",
+        });
         if (googlevideoResponse.headers.has("Location")) {
             location = googlevideoResponse.headers.get("Location") as string;
             continue;
         } else {
+            headResponse = googlevideoResponse;
             break;
         }
     }
+    if (headResponse === undefined) {
+        throw new HTTPException(502, {
+            res: new Response(
+                "Google headResponse redirected too many times",
+            ),
+        });
+    }
 
-    const googlevideoResponse = await fetchClient.call(
-        undefined,
-        location,
-        {
+    // =================== REQUEST CHUNKING =======================
+    // if the requested response is larger than the chunkSize, break up the response
+    // into chunks and stream the response back to the client to avoid rate limiting
+    const { readable, writable } = new TransformStream();
+    const stream = new StreamingApi(writable, readable);
+    const googleVideoUrl = new URL(location);
+    const getChunk = async (start: number, end: number) => {
+        googleVideoUrl.searchParams.set(
+            "range",
+            `${start}-${end}`,
+        );
+        const postResponse = await fetchClient(googleVideoUrl, {
             method: "POST",
             body: new Uint8Array([0x78, 0]), // protobuf: { 15: 0 } (no idea what it means but this is what YouTube uses),
             headers: headersToSend,
-        },
+        });
+        await stream.pipe(postResponse.body);
+    };
+
+    const chunkSize = 5_000_000;
```

Comment:
My feeling is that this should probably be configurable.

Reply:
done in https://github.com/iv-org/invidious-companion/pull/95/commits/0f0d056a753405475f3f7c6f8b64c3c4093ef7c4, not sure about the name.

Reply:
Setting this as `networking.video_fetch_chunk_size_mb` for the moment.